### DISCUSSION TITLE
Saving the data

### DISCUSSION THREAD

**Discussion by AustinThoughtLab** [jspsych/jsPsych#815] [Q&A] [answered]
Hi. I've created a task on jsPsych and it's working great so far. In the workshop on Tuesday you mentioned that we should save the data on our computer, not the participant's. Can you explain how to do that?

**Comment by jkhartshorne** [comment, marked as answer]
@AustinThoughtLab, welcome! The answer depends a lot on what you want to do. Basically, you need a database to store the data in. [This page](https://www.jspsych.org/overview/data/) in the jsPsych docs talks about how to communicate with a MySQL database if you have one already set up. This page tells you how to submit the data directly to [Amazon Mechanical Turk](https://www.jspsych.org/overview/mturk/) (obviously only relevant if you are running your study on AMT). 

Either way, you'll also need a Web server to run your experiment on. If you are using AMT, you can probably embed your code right into the default AMT HTML page. There is something similar for Prolific, but I don't know as much about that. 

Another popular option for working with Amazon Mechanical Turk is [PsiTurk](https://psiturk.readthedocs.io/en/stable/), which will help you set up both a server and a database.

However, if you are not working with AMT or Prolific and are going the citizen science route instead a la gameswithwords.org, your best out-of-the-box option is Pushkin, which we'll be talking about in Week 4. We're currently implementing *a lot* of changes right now in preparation for Week 4, so now isn't necessarily a great time to get started with Pushkin, though. (Week 4 will be a great time.)

**Comment by jodeleeuw** [comment]
Another great option is www.cognition.run. It's a new *free* service by @javidalpe et al. and it's really straightforward to use. 

Check out #754 for a discussion.

Also, see #804 for another relevant answer.
